### PR TITLE
 Early detection for votes overflow

### LINKED ISSUE #1391: Implement queue management
Right now as a part of #1365 we drop votes when the voting queue gets full, we should implement a WRED-esque scheme to handle this more gracefully.

### PR DESCRIPTION
**Pull request by SergiySW**
Fixes #1391

### CODE REVIEW

File: rai/node/node.cpp [hunk 1]
```diff
@@ -1899,6 +1953,20 @@ void rai::node::ongoing_rep_crawl ()
 	}
 }
 
+void rai::node::ongoing_rep_calculation ()
+{
+	auto now (std::chrono::steady_clock::now ());
+	vote_processor.calculate_weights ();
+	std::weak_ptr<rai::node> node_w (shared_from_this ());
+	// Repeat in 10 minutes
+	alarm.add (now + std::chrono::seconds (10 * 60), [node_w]() {
```
Comment (cryptocode) [inline rai/node/node.cpp]: Minor: maybe `std::chrono::minutes` here, and the comment can be removed?

File: rai/node/node.cpp [hunk 2]
```diff
@@ -863,7 +863,30 @@ void rai::vote_processor::vote (std::shared_ptr<rai::vote> vote_a, rai::endpoint
 	std::lock_guard<std::mutex> lock (mutex);
 	if (!stopped)
 	{
-		if (votes.size () < 256 * 1024)
+		bool process (rai::rai_network == rai::rai_networks::rai_test_network);
```
Comment (PlasmaPower) [inline rai/node/node.cpp]: This initial value doesn't do anything as it'll be overridden later. I'd put this condition as the first if statement, and keep the rest as "else if"s.

Comment (PlasmaPower) [reply rai/node/node.cpp]: I guess it does help if the rep is >= 5%. Was that intended?

Comment (SergiySW) [reply rai/node/node.cpp]: Test network value required only for votes processor profiling, good catch

File: rai/node/node.cpp (hunk 1 above)
Comment (SergiySW) [reply rai/node/node.cpp]: Sure

**Pull request merged**